subsys/bluetooth/host/smp.c

```diff
@@ -3181,6 +3182,12 @@ static uint8_t smp_pairing_rsp(struct bt_smp *smp, struct net_buf *buf)
 #endif /* CONFIG_BT_SMP_SC_PAIR_ONLY */
 	}
 
+#ifdef CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY
+	/* not reachable, but added to keep compiler happy */
+	return BT_SMP_ERR_AUTH_REQUIREMENTS;
```

Comment:
Zephyr has a `CODE_UNREACHABLE;` macro. I guess the above two lines could be replaced just with that?

Reply:
What was the compile issue? Missing return?

Reply:
I dislike that this function uses `CODE_UNREACHABLE` and `smp_public_key` uses `return BT_SMP_ERR_AUTH_REQUIREMENTS;`. 

Why can't it be consistent?